docs/LLC_CACHE_TEST_PLAN.md

```diff
@@ -0,0 +1,78 @@
+# LLC Cache Test Plan:
+The purpose of this test plan is to provide a comprehensive test case analysis. The LLC Cache Simulator takes a trace file as its input. This trace file has the purpose 
+of simulating Processor Side cache requests, and Bus Operation side requests/traffic. In this sense our LLC must perform many checks to keep the balance of inclusivity and cache coherence
+in the shared memory configuration presented in the project requirements. A single trace may generate an command 0-9 which are summarized below, the preceding sections cover each operation type in detail, 
+along with the Simulation Flow. 
+
+#### Cache Trace Operations Reference Table:
+| Operation Code (n) | Description | Example |
+|-------------------|-------------|----------|
+| 0 | Read request from L1 data cache | `0 10019d94` |
+| 1 | Write request from L1 data cache | `1 10019d88` |
+| 2 | Read request from L1 instruction cache | `2 408ed4` |
+| 3 | Snooped read request | `3 408ed8` |
+| 4 | Snooped write request | `4 408edc` |
+| 5 | Snooped read with intent to modify (RWIM) | `5 408ee0` |
+| 6 | Snooped invalidate command | `6 408ee4` |
+| 8 | Clear cache and reset all state | `8 0` |
+| 9 | Print cache contents | `9 0` |
+
+## Trace Command Coverage: 
+### Command 0 - Read Request from L1 Data Cache
+This is a processor side request from the L1 data cache (i.e. Misses in L1). 
+
+**Simulation Flow:** 
+1. **Address Lookup**
+   - Calculate set index from address
+   - Search all ways in the set for matching tag
+   - Update access pattern for pseudo-LRU
+
+2. **Cache Hit Case**
+   - If line exists in LLC:
+     * Increment hit counter
+     * Increment read counter
+     * Check MESI state of line
+     * If state is Modified or Exclusive:
+         - Send data to L1 (SENDLINE)
+     * If state is Shared:
+         - Send data to L1 (SENDLINE)
+     * Return from operation
+
+3. **Cache Miss Case**
+   - If line doesn't exist in LLC:
+     * Increment miss counter
+     * Increment read counter
```

Comment:
So this means that even if we have to go to DRAM, we consider it a cache read? Is it considered a cache write because we're allocating a new line to our cache?